agent/tcs/model/ecstcs/api.go

```diff
@@ -490,3 +656,71 @@ func (s TaskMetric) String() string {
 func (s TaskMetric) GoString() string {
 	return s.String()
 }
+
+// Validate inspects the fields of the type to determine if they are valid.
+func (s *TaskMetric) Validate() error {
+	invalidParams := request.ErrInvalidParams{Context: "TaskMetric"}
+	if s.ContainerMetrics != nil {
+		for i, v := range s.ContainerMetrics {
+			if v == nil {
+				continue
+			}
+			if err := v.Validate(); err != nil {
+				invalidParams.AddNested(fmt.Sprintf("%s[%v]", "ContainerMetrics", i), err.(request.ErrInvalidParams))
+			}
+		}
+	}
+
+	if invalidParams.Len() > 0 {
+		return invalidParams
+	}
+	return nil
+}
+
+type ULongStatsSet struct {
+	_ struct{} `type:"structure"`
+
+	// Max is a required field
+	Max *int64 `locationName:"max" type:"long" required:"true"`
```

Comment:
there's a mismatch here: https://github.com/moby/moby/blob/master/api/types/stats.go#L111-L114 has uint64.  We'll need to revisit this